File: core-query/src/main/kotlin/ru/patterns/corequery/domain/Account.kt
package ru.patterns.corequery.domain

import ru.patterns.corequery.domain.AccountId
import ru.patterns.corequery.domain.AccountNumber
import ru.patterns.corequery.domain.Balance
import ru.patterns.corequery.domain.ClientId
import java.time.LocalDateTime

/**
 * Счет клиента
 */
data class Account(
    /**
     * Идентификатор счета
     */
    val id: AccountId,
    /**
     * Дата и время открытия счета
     */
    val creationTimestamp: LocalDateTime,
    /**
     * Дата и время блокировки счета. Равен 'null', если счет не заблокирован
     */
    val blockedTimestamp: LocalDateTime?,
    /**
     * Дата и время закрытия счета. Равен 'null', если счет открыт
     */
    val closedTimestamp: LocalDateTime?,
    /**
     * Идентификатор клиента
     */
    val clientId: ClientId,
    /**
     * Номер счета
     */
    val number: AccountNumber,
    /**
     * Баланс счета
     */
    val balance: Balance,
    /**
     * Является ли счет кредитным
     */
    val isCredit: Boolean
)

File: client-interface/src/main/resources/static/client/js/models/Account.js
import {BaseModel} from './BaseModel.js';

export class Account extends BaseModel {

    constructor(data = {}) {
        super(data);
        this.id = data.id || '';
        this.number = data.number || '';
        this.balance = data.balance || 0;
        this.isCredit = data.isCredit || false;
        this.closedTimestamp = data.closedTimestamp || null;
        this.clientId = data.clientId || '';
        this.currency = data.currency || 'RUB';
    }

    isActive() {
        return !this.closedTimestamp;
    }

    getType() {
        return this.isCredit ? 'Кредитный' : 'Дебетовый';
    }

    getStatus() {
        return this.isActive() ? 'Активен' : 'Закрыт';
    }

    getFormattedBalance() {
        const currencySymbols = {
            'RUB': '₽',
            'USD': '$',
            'EUR': '€'
        };
        const symbol = currencySymbols[this.currency] || '₽';
        return `${this.balance} ${symbol}`;
    }
}
